madsim-rdkafka/src/std/client.rs

```diff
@@ -360,17 +360,23 @@ impl<C: ClientContext> Client<C> {
         partition: i32,
         timeout: T,
     ) -> KafkaResult<(i64, i64)> {
+        // XXX: to move the raw pointer into spawn_blocking
+        struct NativePtr(*mut RDKafka);
+        unsafe impl Send for NativePtr {}
+
         let topic_c = CString::new(topic.to_string())?;
-        let native_client = unsafe { NativeClient::from_ptr(self.native_ptr()) };
+        let native_ptr = NativePtr(self.native_ptr());
```

Comment:
It does not seems to be safe if we cancel the future for `spawn_blocking`. 😕 We may require an `Arc<Self>` here considering that `StreamConsumer` holds an `Arc` over `BaseConsumer`.